aquadoggo/src/tests.rs

```diff
@@ -0,0 +1,402 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+//! E2E test and mini-tutorial for aquadoggo.
```

Comment:
This is very beautiful! :rose: 

I've renamed the file to `tests.rs` as the place of the file in the root already indicates that it is "end-to-end".